mbtools/models.py

```diff
@@ -163,6 +163,36 @@ def inject_question_uuids(self):
                 if not utils.validate_uuid4(id_number[0].text):
                     id_number[0].text = str(uuid4())
 
+    def validate_question_uuids(self):
```

Comment:
Per related comments, I think the business logic of validation should be moved to `validate_mbz_html.find_qbank_uuid_violations`.  You may want to add a property like `question_bank_entry_id` to `MoodleQuestion` like:

```python
class MoodleQuestion:
    """This class models a <question> from a <question_bank_entry> in
    questions.xml
    """
    def __init__(self, question_elm):
        self.etree = question_elm
        self.id = self.etree.attrib['id']
        self.version = self.etree.xpath("../../version")[0].text
        self.id_number = self.etree.xpath('../../../../idnumber')[0].text
        self.question_type = self.etree.xpath('./qtype')[0].text
        self.question_bank_entry_id = self.etree.xpath("../../../..")[0].attrib["id"]
```

You can then do something like the following to implement validation completely inside of `validate_mbz_html.py`:

```python
def find_qbank_uuid_violations(question_bank):
    questions = question_bank.questions()
    # Build dict of id for each question bank entry
    qbe_to_uuid = {}
    for question in questions:
        # This is to de-duplicate since there might be multiple question versions in one question bank entry
        qbe_to_uuid[question.question_bank_entry_id] = question.id_number
    
    for qbe_id, id_nubmer in qbe_to_uuid.items():
        # Add logic to check for duplicate or invalid UUID value
        pass
```